myserver.cpp

```diff
@@ -0,0 +1,626 @@
+#include "myserver.h"
+#include "IDataBase.cpp"
+#include "IDataBase.h"
+#include "DBErrors.h"
+#include <iostream>
+#include <vector>
+#include <string>
+
+CMyServer::CMyServer(){}
+
+CMyServer::~CMyServer(){}
+
+void CMyServer::startServer(){
+    if(listen(QHostAddress::Any,PORT)){
+        qDebug()<<"Listening for connection";
+    }
+    else{
+        qDebug()<<"Not listening for connection";
+    }
+}
+
+void CMyServer::incomingConnection(qintptr pisocketDescriptor){
+    socket = new QTcpSocket(this);
+    socket->setSocketDescriptor(pisocketDescriptor);
+
+    connect(socket, SIGNAL(readyRead()),this,SLOT(sockReady()));
+    connect(socket, SIGNAL(disconnected()),this,SLOT(sockDisc()));
+
+    qDebug()<<pisocketDescriptor<<"Client connected";
+
+    socket->write("{\"type\":\"connect\",\"status\":\"yes\"}");
+    qDebug()<<"Send client connect status - YES";
+}
+
+void CMyServer::sockReady(){
+    Data = socket->readAll();
+    qDebug()<<"Data: "<<Data;
+    std::vector <std::string> msg;
+
+    nlohmann::json j;
+    j = nlohmann::json::parse(QString(Data).toStdString());
+
+    for (nlohmann::json::iterator it = j.begin(); it != j.end(); ++it) {
+            msg.push_back(it.value());
+    }
+
+        if(msg.at(0)=="AddUser"){
+            CMyServer::m_AddUser(msg);
+        }
+        else if(msg.at(0)=="GetUserId"){
+            CMyServer::m_GetUserId(msg);
+        }
+        else if(msg.at(0)=="GetFriendList"){
+            CMyServer::m_GetFriendList(msg);
+        }
+        else if(msg.at(0)=="AddFriend"){
+            CMyServer::m_AddFriend(msg);
```

Comment:
You don't need to use  CMyServer:: for class methods.
Correct this in all the file.

Reply:
Fixed it